### DISCUSSION TITLE
Chrome browser does not start, in cron. (Ubuntu)

### DISCUSSION THREAD

**Discussion by lunst97** [rubycdp/ferrum#349] [General]
I have a code that downloads a file from a website, it only works with the setting (`headless: false`) otherwise it just doesn't download anything. 
I add the task of running the script to cron and I get this error (`home/peeper/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/ferrum-0.13/lib/ferrum/browser/process.rb:157:in parse_ws_url: Browser did not produce websocket url within 10 seconds, try to increase :process_timeout. See https://github.com/rubycdp/ferrum#customization (Ferrum::Process Timeout Error`)
)

And this error is present only if you run the code from under cron, everything works in a normal console.

Cron: 
eval "$(~/.rbenv/bin/rbenv init - bash)"
DISPLAY=:0
cd /home/peeper/xxx && /home/peeper/.rbenv/versions/2.7.2/bin/ruby main.rb

**Comment by route** [comment]
Meaning that you are running non-interactive shell? https://askubuntu.com/questions/503028/how-to-launch-interactive-shell-script-in-bash-hourly-with-cron

This is definitely not an issue of Ferrum, but rather a question, consider it when you open one.

**Comment by lunst97** [comment]
Everything was generally in a different way, it turns out that it was enough in the ruby script to make all the connected files with an absolute path and everything works without DISPLAY, etc. I suffered for a very long time with this, but everything turned out to be on the surface.

**Reply by route** [reply]
Haha true, but definitely hard to help with something from my side, as it's kinda env related.
